cli/src/main.rs

```diff
@@ -52,6 +52,225 @@ pub fn main() {
         }
     };
 
+    let regtest = matches.is_present("regtest");
+    if regtest {
+        use std::ffi::OsString;
+        use std::fs::File;
+        use std::io;
+        use std::path::PathBuf;
+        use std::process::{Command, Stdio};
+        use std::{thread, time};
+
+        // confirm we are in running from zingolib as an anchor for our directory context
+        let revparse = Command::new("git")
+            .args(["rev-parse", "--show-toplevel"])
+            .output()
+            .expect("no git!? time to quit.");
+
+        let revparse_str = std::str::from_utf8(&revparse.stdout).expect("revparse_str error");
+
+        //if revparse ends in zingolib, use that as the start to build the dir
+        if revparse_str.trim_end().ends_with("zingolib") {
```

Comment:
You check before to see that you're in the expected location with respect to the git repo (line 65 of this commit).
If I Understand Correctly (IIUC) this if statement is forcing the developer to name their repo `zingolib`.
Why is that?

Reply:
Line 65 starts a function that will return the top level of whatever repo it is in.

the `zingolib` is an attempt to find a base directory that is actually a clone of `zingolib`. You're right though, it does limit the developer's ability to rename their directory.

I'll look at this again to see if it could be checked with other things (like the file check, for example) - making this un-needed.

Good note!